### PR TITLE
API for subscribing to multicast addresses

### PR DESCRIPTION
**Pull request by tiniuclx**

### CODE REVIEW

File: port/linux/ipadapter.c [hunk 1]
```diff
@@ -1884,8 +1884,51 @@ ip_context_rfds_fd_copy(ip_context_t *dev)
 }
 
 void
-oc_connectivity_subscribe_mcast_ipv6(size_t device, const uint8_t *address)
+oc_connectivity_subscribe_mcast_ipv6(oc_endpoint_t *address)
```
Comment (WAvdBeek) [inline port/linux/ipadapter.c:1887]: missing size_t device_index ?

File: port/oc_connectivity.h [hunk 2]
```diff
@@ -210,11 +210,11 @@ void handle_session_event_callback(const oc_endpoint_t *endpoint,
...
  *
- * @param device ID of device to subscribe
- * @param address 16-byte array containing multicast address
+ * @param address endpoint describing the address to subscribe to.
+ * The device & addr.ipv6.address members must be set for the
+ * function call to be valid.
  */
-void oc_connectivity_subscribe_mcast_ipv6(size_t device,
-                                          const uint8_t *address);
+void oc_connectivity_subscribe_mcast_ipv6(oc_endpoint_t *address);
```
Comment (WAvdBeek) [inline port/oc_connectivity.h:217]: same here

File: port/windows/ipadapter.c [hunk 3]
```diff
@@ -1756,8 +1756,50 @@ oc_dns_lookup(const char *domain, oc_string_t *addr, enum transport_flags flags)
 #endif /* OC_DNS_LOOKUP */
 
 void
-oc_connectivity_subscribe_mcast_ipv6(size_t device, const uint8_t *address)
+oc_connectivity_subscribe_mcast_ipv6(size_t device, oc_endpoint_t *address)
```
Comment (WAvdBeek) [inline port/windows/ipadapter.c:1759]: or remove device here

File: port/linux/ipadapter.c (hunk 1 above)
Comment (tiniuclx) [reply port/linux/ipadapter.c:1887]: Yes, the endpoint has the device information. The doxygen comment for the API clarifies this.

**Review by WAvdBeek — APPROVED** [review]

**Pull request merged**